### PR TITLE
chore: add kubefed fork

### PR DESCRIPTION
**Pull request by hectorj2f**
Signed-off-by: Hector Fernandez <hectorj@gmail.com>

**What type of PR is this?**

Chore
**What this PR does/ why we need it**:

While we are waiting to get https://github.com/kubernetes-sigs/kubefed/pull/1377 merged, we need to workaround that delay by getting our internal kubefed version based on v0.5.1.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

**Comment by jimmidyson** [comment]
We can get https://github.com/kubernetes-sigs/kubefed/pull/1377 merged quickly (like tomorrow morning) I'm sure. Better than using a fork here imo.

File: staging/kubefed/charts/controllermanager/crds/crds.yaml [hunk 1]
```diff
@@ -0,0 +1,1157 @@
...
+                description: DisabledTLSValidations defines a list of checks to ignore
+                  when validating the TLS connection to the member cluster.  This
+                  can be any of *, SubjectName, or ValidityPeriod. If * is specified,
+                  it is expected to be the only option in list.
+                items:
+                  type: string
+                type: array
+              proxyURL:
+                description: ProxyURL allows to set proxy URL for the cluster.
+                type: string
```
Comment (mhrabovcin) [inline staging/kubefed/charts/controllermanager/crds/crds.yaml:397]: This field is the reason we're adding fork of `kubefed` chart.

**Review by mhrabovcin — APPROVED** [review]
This will unblock https://jira.d2iq.com/browse/D2IQ-74314

**Comment by mhrabovcin** [comment]
@jimmidyson the upstream PR is against `master` branch. As @hectorj2f mentioned `yakcl` upgrade of `kubefed` would be non-trivial and it is not in the scope of our unidirectional work.

With merging this PR we'll get unblocked by temporarily using forked kubefed chart + custom built kubefed `v0.5.2` containers. Once we have confidence in delivering unidirectional feature on time we'll invest time in upstream PR and bumping `yakcl` to use latest `kubefed`.

File: staging/kubefed/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v2
+description: KubeFed helm chart
+name: kubefed
+version: 0.5.2
```
Comment (joejulian) [inline staging/kubefed/Chart.yaml:4]: Please don't match the chart version to the app version. These should be expected to diverge and when they do it causes confusion.

This is true for all Chart.yaml in this PR.

**Review by joejulian — CHANGES_REQUESTED** [review]

**Review by joejulian — COMMENTED** [review]
I see that this is only a temporary fork so I'll approve despite my CR.

**Review by joejulian — APPROVED** [review]

**Review by gracedo — APPROVED** [review]
This is only pulled in for yakcl right? Just wondering if it's easier to pull the chart directly into the `kommander-federation` local charts dir instead of using an extra m/charts fork, esp if this is just temporary

**Comment by mhrabovcin** [comment]
@gracedo I don't think we've considered this option. But you are right, this is temporary solution and the chart will be removed from our charts repo.

**Pull request merged**